### PR TITLE
Add hb_creative_load_method keyword to winning bid.AdServerTargeting in PBSResponse

### PR DESCRIPTION
**Pull request by nhedley**
We need a new ad server targeting keyword `hb_creative_load_method` to be returned for the winning bid for Prebid Mobile. This keyword will tell us how we need to load the creative, either in html or in native code by the demand SDK. Needed specifically for Facebook demand right now.

Also abstracted out sortBidsAddKeywords functionality to make it easier to unit test and to make the code cleaner.

### CODE REVIEW

File: pbs_light.go [hunk 1]
```diff
@@ -462,6 +413,63 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
...
+			}
+			// For the top bid, we want to add the following additional keys
+			if i == 0 {
+				pbs_kvs["hb_pb"] = roundedCpm
+				pbs_kvs["hb_bidder"] = bid.BidderCode
+				pbs_kvs["hb_cache_id"] = bid.CacheID
+				if bid.BidderCode == "audienceNetwork" {
+					pbs_kvs["hb_creative_load_method"] = "demand_sdk"
+				} else {
+					pbs_kvs["hb_creative_load_method"] = "html"
```
Comment (dbemiller) [inline pbs_light.go]: Can we turn these magic strings into constants?

All that stuff you explained in slack would (if you can make it a bit more concise) be great to have in go-docs.

**Comment by dbemiller** [comment]
Looks pretty good to go for me. Thanks for helping to carve `auction()` up, and adding tests!

Just needs to fix merge conflicts.

File: pbs_light.go [hunk 2]
```diff
@@ -96,7 +96,14 @@ type bidResult struct {
 	bid_list pbs.PBSBidSlice
 }
 
-const DEFAULT_PRICE_GRANULARITY = "med"
+const defaultPriceGranularity = "med"
+
+const hbCreativeLoadMethodDemandSDK = "demand_sdk"
+const hbCreativeLoadMethodHTML = "html"
```
Comment (dbemiller) [inline pbs_light.go]: Could you leave some comments describing the purpose of these two?

The info you gave me about `demand_sdk` being used to target bidders who insist on control over the webview,  and `html` being used as the default, seem pretty important.

I guess all I'm hoping for is... if we stumble on this code in 3 years and wonder "do we still need this?" We'll need to know what purpose it serves in the first place.

File: pbs_light.go [hunk 3]
```diff
@@ -440,6 +403,66 @@ func auction(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
...
+
+	// record bids by ad unit code for sorting
+	code_bids := make(map[string]pbs.PBSBidSlice, len(bids))
+	for _, bid := range bids {
+		code_bids[bid.AdUnitCode] = append(code_bids[bid.AdUnitCode], bid)
+	}
+
+	// loop through ad units to find top bid
+	for _, unit := range pbs_req.AdUnits {
+		bar := code_bids[unit.Code]
```
Comment (mkendall07) [inline pbs_light.go]: why is this called `bar` ?

Comment (dbemiller) [reply pbs_light.go]: probly written by people with too little time to spare...

she's just moving this block of code, though--not writing it from scratch... so i don't think it should hold up the merge.

**Pull request merged**